### PR TITLE
"Add responsive navigation bar with Home, Contact, About, and Courses links"     "Implement filter tags and categories functionality"   "Create a responsive layout for improved user experience"     "Fix bug in email sending functionality"

### PR DESCRIPTION
**Pull request by HourVireakboth**

### CODE REVIEW

File: src/Components/contact/ContactsForm.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      .min(10, "Your name must 10 letter up"),
+  });
+
+  const handleRegister = async (values, actions) => {
+    const contact = {
+      name: values.name,
+      email: values.email,
+      subject: values.comment,
+      comment: values.message,
+    };
```
Comment (BunHouth) [inline src/Components/contact/ContactsForm.js]: doesn't need to restructure the value, better organize the state value

File: src/Components/contact/ContactsForm.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    setIsLoading(false);
```
Comment (BunHouth) [inline src/Components/contact/ContactsForm.js]: isLoading should be true when start progress

File: src/Components/courses/CourseForm.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        const response = await fetchTags();
+        if (response.results) {
+          console.log(response.results);
+          setTags(response.results);
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    DataTags();
```
Comment (BunHouth) [inline src/Components/courses/CourseForm.js]: function better move to outside the hook.

Comment (BunHouth) [reply src/Components/courses/CourseForm.js]: please using the proper function name

File: src/Components/courses/CourseForm.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        const response = await fetchCourses(selectedCategories, selectedTags);
+        if (response.results) {
+          console.log(response.results);
+          setCourse(response.results);
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    DataCourse();
```
Comment (BunHouth) [inline src/Components/courses/CourseForm.js]: same above

**Review by BunHouth — COMMENTED** [review]
Please check

File: src/Components/Dropdown.js [hunk 5]
```diff
@@ -1,11 +1,43 @@
...
+        onClick={() => setIsOpen((pre) => !pre)}
+        className={` ${
+          isOpen ? "rounded-t-lg" : "rounded-lg"
+        } bg-gray-200 p-4 w-full flex items-center justify-between font-bold text-blue-700 text-lg -tracking-wider border-2 border-transparent border-slate-300  `}
+      >
+        {label}
+        {!isOpen ? (
+          <FontAwesomeIcon icon={faCaretDown} />
+        ) : (
+          <FontAwesomeIcon icon={faCaretUp} />
```
Comment (BunHouth) [inline src/Components/Dropdown.js]: using ternary operator to switch `faCaretDown` and `faCaretUp` instead

File: src/Components/courses/Courses.js [hunk 6]
```diff
@@ -2,25 +2,23 @@
...
 const Courses = () => {
   const [courses, setCourses] = useState([]);
   const [categories, setCategories] = useState([]);
   const [tags, setTags] = useState([]);
   const [selectCategory, setSelectCategory] = useState([]);
   const [selectTag, setSelectTag] = useState([]);
-  const [data, setData] = useState([]);
-  // const []
 
   useEffect(() => {
```
Comment (BunHouth) [inline src/Components/courses/Courses.js:48]: function should define outside the useEffect

File: src/Components/login/FormLogIn.js [hunk 7]
```diff
@@ -20,29 +20,47 @@ const FormLogIn = () => {
...
+    setIsLoading(true);
     try {
       const response = await userLogin(user);
       if (response.token) {
         console.log(response.token);
         setCookies(response.token);
         setSession({ token: response.token });
         router.push("/home");
-        // actions.formReset();
+        actions.resetForm();
```
Comment (BunHouth) [inline src/Components/login/FormLogIn.js:37]: action should be set before push router

File: src/Components/login/FormLogIn.js [hunk 8]
```diff
@@ -20,29 +20,47 @@ const FormLogIn = () => {
...
+
+  const handleLoginGoogle = async () => {
+    try {
+      const response = await userGoogle();
+      if (response.token) {
+        console.log(response.token);
+        setCookies(response.token);
+        setSession({ token: response.token });
+        router.push("/home");
+        actions.resetForm();
```
Comment (BunHouth) [inline src/Components/login/FormLogIn.js:55]: same above

File: src/actions/Courses.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { api } from "@/utils/api";
+export const fetchCourses = async (selectedCategory, selectTags) => {
+  const response = await api.get("/courses?pages=0&pageSize=50", {
```
Comment (BunHouth) [inline src/actions/Courses.js:3]: why page already 0? and page size is 50?

**Comment by HourVireakboth** [comment]
Sorry b this is the old code I use for testing pagination so I put 0 as
default but I have alr refactor it into dynamic Page and page size but
haven’t push yet

File: src/utils/api.js [hunk 10]
```diff
@@ -2,3 +2,7 @@ import axios from "axios";
 export const api = axios.create({
   baseURL: "https://course-web-service.onrender.com/api/v1",
 });
+
+export const apiEmail = axios.create({
+  baseURL: "https://courses-web-service-api-v1.onrender.com/api/v1",
+});
```
Comment (BunHouth) [inline src/utils/api.js:8]: why there are 2 api?
Using endpoint from .env

**Review by BunHouth — COMMENTED** [review]
Please check again.

**Comment by HourVireakboth** [comment]
Because the first end point I use from Marwin api to send contact mail and
the other one from Yuth for user authentication

File: tsconfig.json
Comment (BunHouth) [inline tsconfig.json:1]: project using JS, why you need this kind of typescript config file?

File: src/utils/toDate.js [hunk 11]
```diff
@@ -0,0 +1,6 @@
+export function toDate(date) {
+    const format = new Date(date)
+
+    console.log(format)
+    return format.getDate() +"-" + format.getMonth() + 1 +"-" + format.getFullYear()
+}
```
Comment (BunHouth) [inline src/utils/toDate.js:6]: using dayjs lib instead

**Review by BunHouth — COMMENTED** [review]
please check

**Pull request merged**